doc/api/crypto.md

```diff
@@ -1044,6 +1044,76 @@ encoding of `'utf8'` is enforced. If `data` is a [`Buffer`][], `TypedArray`, or
 
 This can be called many times with new data as it is streamed.
 
+## Class: KeyObject
+<!-- YAML
+added: REPLACEME
+-->
+
+Node.js uses an internal `KeyObject` class which should not be accessed
+directly. Instead, factory functions exist to create instances of this class
+in a secure manner, see [`crypto.createSecretKey`][],
+[`crypto.createPublicKey`][] and [`crypto.createPrivateKey`][]. A `KeyObject`
+can represent a symmetric or asymmetric key, and each kind of key exposes
+different functions.
+
+Most applications should consider using the new `KeyObject` API instead of
+passing keys as strings or `Buffer`s due to improved security features.
+
+### keyObject.getType()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {string}
+
+Depending on the type of this `KeyObject`, this function either returns
+`'secret'` for symmetric keys, `'public'` for public (asymmetric) keys or
+`'private'` for private (asymmetric) keys.
+
+### keyObject.getSymmetricSize()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {number}
+
+For symmetric keys, this function returns the size of the key in bytes. This
+function is undefined for asymmetric keys.
```

Comment:
It would be a completely different measure though. This seems to be related to our discussion around exposing properties of the key (such as the RSA modulus etc) via this API, maybe we should move forward in that direction?

Reply:
Why is it a completely different measure?

Reply:
Oh, I meant "completely different from symmetric sizes", sorry for the confusion. It might make sense, but we should probably think about use cases... Would it make sense to represent this in bits, since the size of a bigint us not necessarily a multiple of eight bits?